services/group-service.ts

```diff
@@ -37,18 +37,24 @@ export default class GroupService implements GroupServiceI{
      * body formatted as JSON containing the new group that was inserted in the
      * database
      */
-    createGroup = (req: Request, res: Response) => {
+    createGroup = async (req: Request, res: Response) => {
         // @ts-ignore
         const creatorUid = req.params.uid === "me" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
-        const usersAddedIds = req.body.usersAddedIds;
-        const allUids = usersAddedIds.append(creatorUid);
 
-        if(creatorUid === "me" || !usersAddedIds){
+        if(creatorUid === "me" || !req.body.users){
             res.sendStatus(503);
             return;
         }
+        req.body.users.push(creatorUid);
+
         try {
-            GroupService.groupDao.createGroup(creatorUid, allUids, req.body)
+            const existingGroups = await GroupService.groupDao.findAllUserGroups(creatorUid);
+            const filteredGroups = existingGroups.filter(group => group.name.toLowerCase() === req.body.group.name.toLowerCase());
```

Comment:
Let's discuss this again. I thought this and later realized that we are making a mistake here.